src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -523,20 +532,25 @@ reactor inp commandMap = do
 
         ReqCodeAction req -> do
           liftIO $ U.logs $ "reactor:got CodeActionRequest:" ++ show req
-          handleCodeActionReq tn commandMap req
+          handleCodeActionReq tn req
           -- TODO: make context specific commands for all sorts of things, such as refactorings          
 
         -- -------------------------------
 
         ReqExecuteCommand req -> do
           liftIO $ U.logs $ "reactor:got ExecuteCommandRequest:" ++ show req
           let params = req ^. J.params
-              command' = params ^. J.command
-              -- if this is a UUID then use the mapping for it
-              command = fromMaybe command' (BM.lookupR command' commandMap)
+              command = stripCmdPrefix (params ^. J.command)
+
+              -- | strips the PID prefix from the command id if it has one
+              stripCmdPrefix :: T.Text -> T.Text
+              stripCmdPrefix x
+                | T.count ":" x >= 2 = T.tail $ T.dropWhile (/= ':') x
+                | otherwise = x
+
```

Comment:
Maybe just split the command on ":" (using `Data.Text.splitOn`), and if you get three chunks discard the first. Then you will already have broken the command into its plugin and command within the plugin.

Reply:
Or use something specific for the start of the prefix, and just drop if it is there.

So do something like "eg2:addPragma" becomes "(12345)eg2:addPragma", and if the first char is not an opening paren you keep it as is, else strip off until the closing paren.

Because it is only the vscode client that cares about the prefix, we can toss it when the message comes in.